File: Examples/ClassDiagram/Swift/Player.swift
import Foundation

/// Drives playback over a library. Depends on `Library` and the `Playable` protocol.
public final class Player {
    private let library: Library
    public private(set) var nowPlaying: Playable?

    public init(library: Library) {
        self.library = library
    }

    public func play(_ item: Playable) {
        nowPlaying = item
        item.play()
    }

    public func playFirstItem(inPlaylistNamed name: String) {
        guard let item = library.playlist(named: name)?.items.first else { return }
        play(item)
    }
}

File: Examples/ClassDiagram/Python/player.py
from library import Library
from playable import Playable


class Player:
    """Drives playback over a library. Depends on Library and the Playable interface."""

    def __init__(self, library: Library):
        self.library: Library = library
        self.nowPlaying: Playable | None = None

    def play(self, item: Playable) -> None:
        self.nowPlaying = item
        item.play()

    def playFirstItem(self, playlistNamed: str) -> None:
        playlist = self.library.playlist(playlistNamed)
        if playlist is not None and playlist.items:
            self.play(playlist.items[0])
